<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsFieldDefinitionWidgetBase</class>
 <widget class="QWidget" name="QgsFieldDefinitionWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>303</width>
    <height>346</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="styleSheet">
   <string notr="true">QToolBox::tab:selected {
    font-weight: bold;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinAndMaxSize</enum>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="mFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QToolBox" name="mChooseFieldToolBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="mNewFieldPage">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>287</width>
           <height>266</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="label">
          <string>Create New Layer Field</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,0,1">
          <item row="0" column="1" colspan="3">
           <widget class="QLineEdit" name="mFieldNameLineEdit"/>
          </item>
          <item row="2" column="1">
           <widget class="QgsSpinBox" name="mLengthSpinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="mFieldNameLabel">
            <property name="text">
             <string>Name</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QgsSpinBox" name="mPrecisionSpinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="mTypeLabel">
            <property name="text">
             <string>Type</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="mLengthLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Length</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="3">
           <widget class="QComboBox" name="mTypeCmb"/>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="mPrecisionLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Precision</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="4">
           <widget class="QgsCollapsibleGroupBox" name="mAdvancedGroup">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Advanced Fields</string>
            </property>
            <property name="saveCheckedState">
             <bool>true</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
             <item row="0" column="0">
              <widget class="QLabel" name="mAliasLabel">
               <property name="text">
                <string>Alias</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="mCommentLineEdit">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="mAliasLineEdit">
               <property name="toolTip">
                <string>Leave empty to use the field name</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="mCommentLabel">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Comment</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QCheckBox" name="mIsNullableCheckBox">
               <property name="text">
                <string>Is Nullable</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="mExistingFieldPage">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>266</width>
           <height>44</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="label">
          <string>Use Existing Layer Field</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
          <item row="0" column="1">
           <widget class="QgsFieldComboBox" name="mExistingLayerFieldCmb"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="mExistingLayerFieldLabel">
            <property name="text">
             <string>Field</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
